crypto/txscript/src/lib.rs

```diff
@@ -973,6 +1047,231 @@ mod tests {
             );
         }
     }
+
+    #[derive(Clone)]
+    struct SignatureData {
+        signature: Vec<u8>,
+        public_key: Vec<u8>,
+    }
+
+    enum SignatureScriptBuilder {
+        Single(SignatureData),
+        Multi(Vec<SignatureData>),
```

Comment:
I think Multisig gives a clearer context. And also comment on each variant would be nice

Reply:
dfc20f7043ceb6eb79633905b41a2a436b0031e5